torch/_inductor/runtime/triton_helpers.py

```diff
@@ -198,25 +198,47 @@ def any(a, dim):
 
 @triton.jit
 def bucketize_binary_search(
-    values,  # 1D tensor
-    offsets_ptr,
-    indexing_dtype,
+    values: tl.tensor,
+    boundaries_ptr: tl.tensor,
+    BOUNDARIES_SIZE: tl.constexpr,
+    BOUNDARIES_UNDERLYING_NUMEL: tl.constexpr,
+    BOUNDARIES_STRIDE: tl.constexpr,
+    boundary_indices: tl.tensor,  # Indices pointing to the beginning of the 1-D sequence of offsets used to bucketize each value
+    indexing_dtype: tl.dtype,
     right,  # bool: if true, use intervals closed on the left; see [Note: Inductor bucketize op]
-    OFFSETS_SIZE: int,
+    sorter_ptr: tl.tensor,  # optional pointer of the same shape as offsets_ptr, None if not present
+    SORTER_STRIDE: tl.constexpr,
+    sorter_indices: tl.tensor,
     BLOCK_SHAPE,  # tuple/list of block shape
```

Comment:
Could you add type annotations for `BLOCK_SHAPE`, and `right`?

Reply:
It doesn't look like I can; `triton` barfs on parsing type annotations it doesn't understand, and I can't find a suitable version of these that it does understand.

Reply:
What have you tried. I think `tl.constexpr` would fit here, no?

Reply:
Does `tl.constexpr` apply to something with `len() > 1`?

Reply:
Ah! `BLOCK_SHAPE != BLOCK_SIZE` it should work for `right` though?

Reply:
Weirdly no. `triton` didn't understand the `bool` type. I was just as surprised as you. :sweat_smile:

Reply:
Sorry, I mean giving `right` a `tl.constexpr`. Triton should be fine with that